### PR TITLE
add some recursive functions to existing tools

### PR DESCRIPTION
**Pull request by adclose**
add recursive mkdir
add upload_file force (to make directory if not there)
add recursive to list function

### CODE REVIEW

**Comment by adclose** [comment]
I didn't have the credentials to run the tests on the merge.  if someone else could run the tests and verify that all pass.

**Comment by ezhov-evgeny** [comment]
@adclose Hi! Thanks for interest! I have written for you small instruction:

### Prepare development environment
1. Install docker on your development machine
1. Start WebDAV server for testing by following commands from the project's root folder or change path to `conf` dir in second command to correct:
```shell script
docker pull bytemark/webdav
docker run -d --name webdav -e AUTH_TYPE=Basic -e USERNAME=alice -e PASSWORD=secret1234 -v conf:/usr/local/apache2/conf -p 8585:80 bytemark/webdav
``` 

### Code convention

Please check your code according PEP8 Style guides.

### Run tests
1. Check that webdav container is started on your local machine
1. Execute following command in the project's root folder:
```shell script
python -m unittest discover -s tests
```

### Prepare a Pull Request

Please use this check list before creating PR:
1. You code should be formatted according PEP8
1. All tests should successfully pass
1. Your changes shouldn't change previous default behaviour, exclude defects

File: webdav3/client.py [hunk 1]
```diff
@@ -234,12 +234,13 @@ def valid(self):
         return True if self.webdav.valid() else False
 
     @wrap_connection_error
-    def list(self, remote_path=root, get_info=False):
+    def list(self, remote_path=root, get_info=False, recursive=True):
```
Comment (ezhov-evgeny) [inline webdav3/client.py:237]: `recursive` should have `False` value by default for not changing previous behaviour.

File: webdav3/client.py [hunk 2]
```diff
@@ -251,12 +252,15 @@ def list(self, remote_path=root, get_info=False):
...
+        headers = []
+        if recursive == True:
+            headers = ["Depth:infinity"]
         directory_urn = Urn(remote_path, directory=True)
         if directory_urn.path() != Client.root and not self.check(directory_urn.path()):
             raise RemoteResourceNotFound(directory_urn.path())
 
         path = Urn.normalize_path(self.get_full_path(directory_urn))
-        response = self.execute_request(action='list', path=directory_urn.quote())
+        response = self.execute_request(action='list', path=directory_urn.quote(),headers_ext=headers)
```
Comment (ezhov-evgeny) [inline webdav3/client.py:263]: Please add space after `,` according PEP8

File: webdav3/client.py (hunk 1 above)
Comment (ezhov-evgeny) [inline webdav3/client.py:237]: Also we need to write tests for new functionality otherwise Quality gate will be failed

**Comment by adclose** [comment]
I made the code style updates, I'm not well versed in testing so I didn't add that.  If you have a code example or time to explain the testing I would be willing to add them too.

**Review by ezhov-evgeny — APPROVED** [review]

**Pull request merged**